src/reader/state.rs

```diff
@@ -114,21 +114,19 @@ impl ReaderState {
                 )))
             }
             BangType::DocType if uncased_starts_with(buf, b"!DOCTYPE") => {
-                let start = buf[8..]
-                    .iter()
-                    .position(|b| !is_whitespace(*b))
-                    .unwrap_or(len - 8);
-                if start + 8 >= len {
-                    // Because we here, we at least read `<!DOCTYPE>` and offset after `>`.
-                    // We want report error at place where name is expected - this is just
-                    // before `>`
-                    self.offset -= 1;
-                    return Err(Error::IllFormed(IllFormedError::MissedDoctypeName));
+                match buf[8..].iter().position(|&b| !is_whitespace(b)) {
+                    Some(start) => Ok(Event::DocType(BytesText::wrap(
+                        &buf[8 + start..],
```

Comment:
It would be kind of nice if we could document these constants here and above.  With a bit of thought most of them are not so difficult to figure out, but they're not obvious at first clance.

e.g.

```rust
            BangType::DocType if uncased_starts_with(buf, b"!DOCTYPE") => {
                let doctype_tag_len = 8;
                match buf[doctype_tag_len..].iter().position(|&b| !is_whitespace(b)) {
                    Some(start) => Ok(Event::DocType(BytesText::wrap(
                        &buf[doctype_tag_len + start..],

```

Feel free to tweak the details (such as `const` vs `let`, `doctype_tag_len` vs `DOCTYPE_TAG.len()`)

Reply:
Lines 87 - 113 could use the same treatment.

Reply:
Yes, that code already slightly rewritten (I'll push draft PR soon) and does not contain magic numbers (or they documented)